c2/cuda/groth16_srs.cuh

```diff
@@ -178,20 +177,33 @@ private:
                 } h, l, a, b_g1, b_g2;
             } data;
 
+            if (!ngpus()) {
+                throw sppark_error{CUDA_ERROR_NO_DEVICE,
+                                   std::string("No CUDA devices available")};
```

Comment:
Just in case, `std::string` is not actually necessary, passing a normal C string works just fine :-)